src/hotspot/share/prims/jvmtiRedefineClasses.cpp

```diff
@@ -991,29 +1008,43 @@ jvmtiError VM_RedefineClasses::compare_and_normalize_class_versions(
   JavaFieldStream old_fs(the_class);
   JavaFieldStream new_fs(scratch_class);
   for (; !old_fs.done() && !new_fs.done(); old_fs.next(), new_fs.next()) {
-    // access
-    old_flags = old_fs.access_flags().as_short();
-    new_flags = new_fs.access_flags().as_short();
-    if ((old_flags ^ new_flags) & JVM_RECOGNIZED_FIELD_MODIFIERS) {
-      return JVMTI_ERROR_UNSUPPORTED_REDEFINITION_SCHEMA_CHANGED;
-    }
-    // offset
-    if (old_fs.offset() != new_fs.offset()) {
-      return JVMTI_ERROR_UNSUPPORTED_REDEFINITION_SCHEMA_CHANGED;
-    }
```

Comment:
You've changed the ordering of the checks. I'm not sure why.

Reply:
I changed the order to report about different names first if any of the variables are added/deleted. And report about different modifiers/offsets only if the names of variables are the same.
The result should be the same because JVMTI_ERROR_UNSUPPORTED_REDEFINITION_SCHEMA_CHANGED is returned anyway.